subsys/bluetooth/mesh/adv.c

```diff
@@ -40,8 +40,9 @@ const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES] = {
 	[BT_MESH_ADV_URI]    = BT_DATA_URI,
 };
 
-static K_FIFO_DEFINE(bt_mesh_adv_queue);
-static K_FIFO_DEFINE(bt_mesh_relay_queue);
+static K_QUEUE_DEFINE(bt_mesh_adv_queue);
+static K_QUEUE_DEFINE(bt_mesh_relay_queue);
```

Comment:
Though we are planning to add a separate adv set for Friend for better friendship performance, it is actually good idea to use `K_QUEUE` as we could use this to put Friend Update messages in front of the local queue so they are not delayed by other local messages when 1 adv set is used for local messages and friendship.